### DISCUSSION TITLE
Can I use a ProjectID on a local server?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1575] [Developer Support]
There are two versions of this question:

"When i use a local relayer server, Can i use SignClient.init without  projectId or how to make projectId on local server ?"

"Does a "hello world" example of localhost testing exist anywhere? I have yet to find a single example of relayUrl that does not use relay.walletconnect.com. "

Is there a way to do this? Or is there a reason why using relay.walletconnect.com/projectID... shouldn't be an issue for local development?

**Comment by arein** [comment]
A "local" or "self-hosted" server means running a websocket server that speaks the WalletConnect protocol yourself. We've open-sourced the initial server that served the alpha version of WalletConnect v2 [here](https://github.com/WalletConnect/ts-relay).

We currently use project ids to identify developers and usage insights around them e.g. are the client SDK versions used up to date. Access to our project id server is not public.

Generally we do not recommend "self-hosting" right now as the self-hosted server wouldn't be interoperable with the wider network.

We are working hard to enable this and we just [raised a new funding round](https://medium.com/walletconnect/walletconnect-raises-12-5-million-to-build-out-a-decentralized-communications-network-9f1851709954) to this purpose.

So my recommendation is to hold-off running your own server in prod until we release a new version of the server.

**Comment by amp2122** [comment]
For anyone reading this later, here is the new link to the example repo that was linked in the reply above: https://github.com/WalletConnect/relay

Next to the [v1 node bridge server implementation](https://github.com/WalletConnect/node-walletconnect-bridge), there exists an alternative [v1 java bridge server implementation](https://github.com/aktionariat/walletconnect-bridge).

It is important that an easily deployable bridge server implementation exists, which does not rely on WC infra.
